exercises/static/exercises/dl_digit_classifier_newmanager/python_template/ros2_humble/HAL.py

```diff
@@ -0,0 +1,41 @@
+import rclpy
+from rclpy.node import Node
+from sensor_msgs.msg import Image
+from cv_bridge import CvBridge
+import threading
+import cv2
+
+current_frame = None  # Global variable to store the frame
+
+class WebcamSubscriber(Node):
```

Comment:
Use the CameraNode HAL interface

Reply:
Trying to use the CameraNode HAL interface was unsuccessful in capturing frames due to the physical camera setup. Consequently, I had to switch to a separate subscriber code.

Reply:
um but this is just a subscriber to image_raw and not the publisher right? The publisher node can be the one you used in the launch file, but I think the subscriber can be changed to the one from HAL interfaces

Reply:
Yes the publisher node is present in the launch file, but using the HAL interface as subscriber for this particular package was not able to obtain the image feed, hence we had to resort to a subscriber in HAL.py